File: android/beacon/1-QuickStart/Complete/app/src/main/java/android/connecthings/com/tuto/quickstart/MyNotificationFilter.java
package android.connecthings.com.tuto.quickstart;

import android.support.annotation.NonNull;

import com.connecthings.connectplace.actions.model.PlaceNotification;
import com.connecthings.connectplace.actions.notification.filters.NotificationFilter;
import com.connecthings.connectplace.common.utils.dataholder.DataHolder;

/**
 * Created by Connecthings on 07/11/2017.
 */
public class MyNotificationFilter implements NotificationFilter {
    private static final String MIN_NEXT_TIME_NOTIFICATION = "com.tuto.notification.minNextTimeNotification";
    private long minNextTimeNotification;
    private int minTimeBetweenNotification;

    public MyNotificationFilter(int minTimeBetweenNotification) {
        this.minTimeBetweenNotification = minTimeBetweenNotification;
    }

    @Override
    public String getName() {
        return MyNotificationFilter.class.getSimpleName();
    }

    @Override
    public boolean createNewNotification(@NonNull PlaceNotification placeNotification) {
        return minNextTimeNotification < System.currentTimeMillis();
    }

    @Override
    public boolean deleteCurrentNotification(@NonNull PlaceNotification placeNotification) {
        return true;
    }

    @Override
    public void onNotificationCreated(@NonNull PlaceNotification placeNotification, boolean b) {

    }

    @Override
    public void onNotificationDeleted(@NonNull PlaceNotification placeNotification, boolean deleted) {
        if (deleted) {
            minNextTimeNotification = System.currentTimeMillis() + minTimeBetweenNotification;
        }
    }

    @Override
    public void onBackground() {

    }

    @Override
    public void onForeground() {

    }

    @Override
    public void load(@NonNull DataHolder dataHolder) {
        minNextTimeNotification = dataHolder.getLong(MIN_NEXT_TIME_NOTIFICATION, 0);
    }

    @Override
    public void save(@NonNull DataHolder dataHolder) {
        dataHolder.putLong(MIN_NEXT_TIME_NOTIFICATION, minNextTimeNotification);
    }

    @Override
    public void updateParameters(@NonNull Object o) {

    }
}

File: ios/SWIFT/Beacon/2-Notification/Complete/Complete/MyNotificationFilter.swift
//
//  MyNotificationFilter.swift
//  Complete
//
//  Created by Connecthings on 24/11/2017.
//  Copyright © 2017 R&D connecthings. All rights reserved.
//

import Foundation

//
//  MyNotificationFilter.swift
//  Complete
//
//  Created by Connecthings on 23/11/2017.
//  Copyright © 2017 R&D connecthings. All rights reserved.
//
import Foundation
import ConnectPlaceActions

class MyNotificationFilter: NotificationFilter {
    static let MIN_NEXT_TIME_NOTIFICATION: String = "com.notification.minNextTimeNotification"

    let minTimeBetweenNotification: Double
    var minNextTimeNotification: Double

    init(_ minTimeBetweenNotification: Double) {
        self.minTimeBetweenNotification = minTimeBetweenNotification
        self.minNextTimeNotification = 0
    }

    public func getName() -> String {
        return "MyNotificationFilter";
    }

    public func createNewNotification(placeNotification: PlaceNotification) -> Bool {
        return self.minNextTimeNotification < CACurrentMediaTime()
    }

    public func deleteCurrentNotification(placeNotification: PlaceNotification?) -> Bool {
        return true
    }

    public func onNotificationCreated(placeNotification: PlaceNotification, created: Bool) {

    }

    public func onNotificationDeleted(placeNotification: PlaceNotification?, deleted: Bool) {
        if (deleted) {
            minNextTimeNotification = CACurrentMediaTime() + minTimeBetweenNotification;
        }
    }

    public func onBackground() {

    }

    public func onForeground() {

    }

    public func load(userDefaults: UserDefaults) {
        minNextTimeNotification = userDefaults.double(forKey: MyNotificationFilter.MIN_NEXT_TIME_NOTIFICATION)
    }

    public func save(userDefaults: UserDefaults) {
        userDefaults.set(minNextTimeNotification, forKey: MyNotificationFilter.MIN_NEXT_TIME_NOTIFICATION)
    }

    public func updateParameters(parameters: AnyObject) {

    }
}

